clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerBackgroundThread.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.clients.consumer.internals;
+
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.clients.GroupRebalanceConfig;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.IsolationLevel;
+import org.apache.kafka.common.utils.KafkaThread;
+import org.apache.kafka.common.utils.LogContext;
+import org.apache.kafka.common.utils.Time;
+import org.slf4j.Logger;
+
+import java.util.Locale;
+import java.util.Optional;
+
+public class ConsumerBackgroundThread<K,V> extends KafkaThread implements AutoCloseable {
+    private static final String CONSUMER_BACKGROUND_THREAD_PREFIX = "consumer_background_thread";
+    private final Time time;
+    private final LogContext logContext;
+    private Logger log;
+    private GroupRebalanceConfig groupRebalanceConfig;
+
+    // Configurations
+    private String clientId;
+    private Optional<String> groupId;
+    private long retryBackoffMs;
+    private int heartbeatIntervalMs;
+    private int requestTimeoutMs;
+    private IsolationLevel isolationLevel;
+
+    // control variables
+    private boolean closed = false;
+
+    public ConsumerBackgroundThread(ConsumerConfig config) {
```

Comment:
We'll probably want to have a constructor that also takes a `Time` object for unit tests.